base/compiler/ssair/passes.jl

```diff
@@ -818,12 +821,17 @@ end
     applyTvar = applyT.var
     applyTbody = applyT.body
 
-    isa(applyTbody, DataType) || return nothing
+    arg = unwrap_unionall(arg)
+    applyTbody = unwrap_unionall(applyTbody)
```

Comment:
```suggestion
    arg = unwrap_unionall(arg)::DataType
    applyTbody = unwrap_unionall(applyTbody)::DataType
```
(how do we know that applyTbody is not a Union?)

Reply:
Good catch, we don't. We need to check.